File: web/src/dtos/Category.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";

export interface CategoryRes {
  categoryId?: number;
  name?: string;
  description?: string;
  code?: string;
  position?: number;
}

export interface CategoryResWithProductsCount extends CategoryRes {
  productsCount?: number;
}

export class CategoryReqEdit {
  name?: string = "";
  description?: string = "";
  code?: string = "";
  position?: number = 10;
}

export class CategoryReqSearch extends PagedReq {
  constructor({
    pageNumber = 1,
    pageSize = Common.DEFAULT_PAGE_SIZE,
    orderBy,
    searchText = "",
  }: PagedReq) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
  }
}

File: data/Dtos/Category.cs
using data.Utility.Paging;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace data.Dtos
{
    public class CategoryRes
    {
        public int CategoryId { get; set; }
        public string? Name { get; set; }
        public string? Code { get; set; }
        public int Position { get; set; }
        public string? Description { get; set; }
    }

    public class CategoryResWithProductsCount : CategoryRes
    {
        public int ProductsCount { get; set; }
    }

    public class CategoryReqEdit
    {
        public string? Name { get; set; }
        public string? Code { get; set; }
        public int Position { get; set; }
        public string? Description { get; set; }
    }

    public class CategoryReqSearch : PagedRequestDto
    {

    }
}
